File: Selenideproject/src/test/java/com/ust/sdet/pages/LoginPage.java
package com.ust.sdet.pages;

import com.ust.sdet.support.Config;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {

    private static final By USERNAME =
            By.cssSelector("[id='email']");

    private static final By PASSWORD =
            By.cssSelector("[id='password']");

    private static final By LOGIN_BUTTON =
            By.cssSelector("button.form-submit");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public LoginPage open() {
        driver.get(Config.baseUrl() + "/login");
        return this;
    }

    public LoginPage enterUsername(String username) {
        type(USERNAME, username);
        return this;
    }

    public LoginPage enterPassword(String password) {
        type(PASSWORD, password);
        return this;
    }

    public DashboardPage clickLogin() {
        click(LOGIN_BUTTON);
        return new DashboardPage(driver);
    }

}

File: Playwright_UI_Testing/pages/LoginPage.ts
import { expect, Page } from "@playwright/test";
import { Config } from "../config/Config";
import { LoginLocators } from "../locators/LoginLocators";
import { logger } from "../logger/Logger";

export class LoginPage {

    private locators: LoginLocators;

    constructor(private page: Page) {
        this.locators = new LoginLocators(page);
    }

    async open() {
        await this.page.goto(Config.baseUrl + "/login");
    }

    async enterUsername(username: string) {
        await this.locators.usernameInput().fill(username);
    }

    async enterPassword(password: string) {
        await this.locators.passwordInput().fill(password);
    }

    async clickLogin() {
        await this.locators.loginButton().click();
    }

    async login(username: string, password: string) {
        logger.info(`Login attempt for user: ${username}`);
        await this.enterUsername(username);
        await this.enterPassword(password);
        await this.clickLogin();
    }

    async verifyLoginPage() {
        await expect(this.locators.usernameInput()).toBeVisible();
        await expect(this.locators.passwordInput()).toBeVisible();
        await expect(this.locators.loginButton()).toBeVisible();
    }
}
